index.js

```diff
@@ -65,47 +66,59 @@ module.exports = function f (b, opts) {
                 outputs = [];
             }
 
+            // utility for creating output streams
             function moreOutputs (file) {
                 if (isarray(outopt)) return [];
                 if (!outopt) return [];
                 var xopts = { env: xtend(process.env, { FILE: file }) };
                 return [ outpipe(outopt, xopts) ];
             }
 
-            var pipelines = files.reduce(function (acc, x, ix) {
+            // create a pipeline (labeled-stream-splicer) for every entry file
+            // with only a packer
+            var pipelines = files.reduce(function (acc, file, index) {
                 var pipeline = splicer.obj([
                     'pack', [ pack(packOpts) ],
                     'wrap', []
                 ]);
 
-                if (ix >= outputs.length) {
-                    outputs.push.apply(outputs, moreOutputs(x));
+                // if not enough specified outputs, possibly add additional outputs (?)
+                if (index >= outputs.length) {
+                    outputs.push.apply(outputs, moreOutputs(file));
                 }
-                if (outputs[ix]) pipeline.pipe(outputs[ix]);
-                
-                acc[path.resolve(cwd, x)] = pipeline;
+                // pipe to output if one exists
+                if (outputs[index]) pipeline.pipe(outputs[index]);
+
+                acc[path.resolve(cwd, file)] = pipeline;
                 return acc;
             }, {});
 
             // Force browser-pack to wrap the common bundle
             b._bpack.hasExports = true;
 
+            // for tests/debugging
```

Comment:
It's also for plugins to hook into factored pipelines. If you have a minification step for example it can be added to the factored pipeline by using this event.